velox/functions/sparksql/tests/ArrayMinTest.cpp

```diff
@@ -0,0 +1,223 @@
+  }
+
+  void testDateNullable() {
+    auto D = [](const std::string& dateStr) { return DATE()->toDays(dateStr); };
+    auto arrayVector = makeNullableArrayVector<int32_t>(
+        {{D("1970-01-01"), D("2023-08-23")},
+         {},
+         {D("1970-01-01"), std::nullopt}},
+        ARRAY(DATE()));
+    auto expected = makeNullableFlatVector<int32_t>(
+        {D("1970-01-01"), std::nullopt, D("1970-01-01")}, DATE());
+    testExpr<int32_t>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testDate() {
+    auto D = [](const std::string& dateStr) { return DATE()->toDays(dateStr); };
+    auto arrayVector = makeArrayVector<int32_t>(
+        {{D("1970-01-01"), D("2023-08-23")}, {}}, DATE());
+    auto expected = makeNullableFlatVector<int32_t>(
+        {D("1970-01-01"), std::nullopt}, DATE());
+    testExpr<int32_t>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testInLineVarcharNullable() {
+    using S = StringView;
+
+    auto arrayVector = makeNullableArrayVector<S>({
+        {S("red"), S("blue")},
+        {std::nullopt, S("blue"), S("yellow"), S("orange")},
+        {},
+        {std::nullopt, std::nullopt},
+        {S("red"), S("purple"), S("green")},
+    });
+    auto expected = makeNullableFlatVector<S>(
+        {S("blue"), S("blue"), std::nullopt, std::nullopt, S("green")});
+    testExpr<S>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testVarcharNullable() {
+    using S = StringView;
+    // use > 12 length string to avoid inlining
+    auto arrayVector = makeNullableArrayVector<S>({
+        {S("red shiny car ahead"), S("blue clear sky above")},
+        {std::nullopt,
+         S("blue clear sky above"),
+         S("yellow rose flowers"),
+         S("orange beautiful sunset")},
+        {},
+        {std::nullopt, std::nullopt},
+        {S("red shiny car ahead"),
+         S("purple is an elegant color"),
+         S("green plants make us happy")},
+    });
+    auto expected = makeNullableFlatVector<S>(
+        {S("blue clear sky above"),
+         S("blue clear sky above"),
+         std::nullopt,
+         std::nullopt,
+         S("green plants make us happy")});
+    testExpr<S>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testBoolNullable() {
+    auto arrayVector = makeNullableArrayVector<bool>(
+        {{true, false},
+         {true},
+         {false},
+         {},
+         {std::nullopt, std::nullopt},
+         {true, false, true, std::nullopt},
+         {std::nullopt, true, false, true},
+         {false, false, false},
+         {true, true, true}});
+
+    auto expected = makeNullableFlatVector<bool>(
+        {false,
+         true,
+         false,
+         std::nullopt,
+         std::nullopt,
+         false,
+         false,
+         false,
+         true});
+    testExpr<bool>(expected, "array_min(C0)", {arrayVector});
+  }
+
+  void testBool() {
+    auto arrayVector = makeArrayVector<bool>(
```

Comment:
The functions are written as simple functions, hence, tests can be simplified and made more readable by testing one value at a time.

```
ASSERT_EQ(arrayMin({true, false}), false);
ASSERT_EQ(arrayMin({}), std::nullopt);
...
```

Here, arrayMin would be a helper template that takes a list of values, converts it into a single-row Array vector, evalaates array_min function and returns an optional result value.

Reply:
rewrite all tests with simple interface like `array_max`